src/data_source/storage/sql.rs

```diff
@@ -184,29 +212,64 @@ fn add_custom_migrations(
         .map(|pair| pair.reduce(|_, custom| custom))
 }
 
-/// Postgres client config.
-#[derive(Clone, Debug)]
-pub struct Config {
-    db_opt: PgConnectOptions,
+pub struct Config<DB>
+where
+    DB: Database,
+{
+    db_opt: <DB::Connection as Connection>::Options,
     pool_opt: PoolOptions<Db>,
+    #[cfg(not(feature = "embedded-db"))]
     schema: String,
+    #[cfg(feature = "embedded-db")]
+    db_path: std::path::PathBuf,
     reset: bool,
     migrations: Vec<Migration>,
     no_migrations: bool,
     pruner_cfg: Option<PrunerCfg>,
     archive: bool,
 }
 
-impl Default for Config {
+#[cfg(not(feature = "embedded-db"))]
+impl Default for Config<Postgres> {
     fn default() -> Self {
         PgConnectOptions::default()
+            .username("postgres")
+            .password("password")
             .host("localhost")
             .port(5432)
             .into()
     }
 }
 
-impl From<PgConnectOptions> for Config {
+#[cfg(feature = "embedded-db")]
+impl Default for Config<sqlx::Sqlite> {
+    fn default() -> Self {
+        SqliteConnectOptions::default()
+            .busy_timeout(Duration::from_secs(30))
+            .auto_vacuum(sqlx::sqlite::SqliteAutoVacuum::Incremental)
+            .into()
```

Comment:
Should we add `create_if_missing` to the default? Seems like a good default behavior to have